### PR TITLE
feat!: arbitrary payload size support

### PR DESCRIPTION
**Pull request by Gozala**
Implements newer Piece hasher / cid as per https://github.com/filecoin-project/FIPs/pull/808#pullrequestreview-1681013969

Tests from spec had been added.

### CODE REVIEW

File: test/lib.spec.js [hunk 1]
```diff
@@ -9,15 +9,49 @@ export const testLib = {
...
 
     assert.deepEqual(
       digest,
       new Uint8Array([
-        145, 32, 33, 2, 55, 49, 187, 153, 172, 104, 159, 102, 238, 245, 151, 62,
-        74, 148, 218, 24, 143, 77, 220, 174, 88, 7, 36, 252, 111, 63, 214, 13,
-        253, 72, 131, 51,
+        145,
+        32, // prefix
+        33, // length
```
[the remark this answers is not in the document]
Comment (Gozala) [reply test/lib.spec.js:20]: Thanks for pointing that out, that does not look right indeed

[the remark this answers is not in the document]
Comment (Gozala) [reply test/lib.spec.js:20]:
```suggestion
        34, // length
```

I wonder what this will do

[the remark this answers is not in the document]
Comment (Gozala) [reply test/lib.spec.js]: Same test in Rust seems to behave as expected https://github.com/web3-storage/fr32-sha2-256-trunc254-padded-binary-tree-multihash/pull/18/commits/59e62bd2f23efe6a8c1144fce1eb76f0dcbd89ec

This is starting to be really interesting

[the remark this answers is not in the document]
Comment (Gozala) [reply test/lib.spec.js]: Ah, I forgot to update wasm bindgen wrapper thingy that hard coded the multihash prefix https://github.com/web3-storage/fr32-sha2-256-trunc254-padded-binary-tree-multihash/pull/18/commits/bd1e8d4d632de4368058e565fa88033a87a4fa6c#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759

**Review by vasco-santos — APPROVED** [review]
From what I can tell this looks great!

**Pull request merged**